docker/tool/linux/Dockerfile.base

```diff
@@ -27,7 +27,11 @@ RUN apt-get update && \
     npm i npm@latest -g && \
     npm i -g azure-iothub yo generator-azure-iot-edge-module && \
     apt-get install -y --no-install-recommends python-dev build-essential libssl-dev libffi-dev libxml2-dev libxslt1-dev zlib1g-dev sudo
-RUN python3 -m pip install --upgrade pip && \
+RUN add-apt-repository ppa:deadsnakes/ppa -y -u && \
```

Comment:
>ppa:deadsnakes/ppa [](http://example.com/codeflow?start=23&length=18)

Is 3.9 version not available in apt? Do we need this dependency?
$ sudo -E add-apt-repository -y 'ppa:deadsnakes/ppa'
 This PPA contains more recent Python versions packaged for Ubuntu.

Disclaimer: there's no guarantee of timely updates in case of security problems or other issues. If you want to use them in a security-or-otherwise-critical environment (say, on a production server), you do so at your own risk.

Reply:
fixed. TY Mariana